### DISCUSSION TITLE
How can I monitor learning rate in pytorch lightning?

### DISCUSSION THREAD

**Discussion by mohanades** [Lightning-AI/pytorch-lightning#16287] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I want to be able to see training model in addition to log and monitor acc and loss,I could be see learning rate . How can I change the following code?

```
def polynomial(base_lr, iter, max_iter = 1e5, power = 1):
    return base_lr * ((1 - float(iter) / max_iter) ** power)

class model(pl.LightningModule):
  def __init__(self, config , learning_rate = 1e-4, max_steps = 100000//2):
    super(model, self).__init__()
    
    self.config = config
    self.save_hyperparameters()
    self.model =  modelf(config)
    self.training_losses = []
    self.validation_losses = []
    self.max_steps = max_steps


  def configure_optimizers(self):
    return torch.optim.AdamW(self.parameters(), lr = self.hparams['learning_rate'])

  def optimizer_step(self, epoch_nb, batch_nb, optimizer, optimizer_i, opt_closure = None):

        ## Warmup for 1000 steps
        if self.trainer.global_step < 1000:
            lr_scale = min(1., float(self.trainer.global_step + 1) / 1000.)
            for pg in optimizer.param_groups:
                pg['lr'] = lr_scale * self.hparams.learning_rate

        ## Linear Decay
        else:
            for pg in optimizer.param_groups:
                pg['lr'] = polynomial(self.hparams.learning_rate, self.trainer.global_step, max_iter = self.max_steps)

        optimizer.step(opt_closure)
        optimizer.zero_grad()
```

**Comment by w2ex** [comment]
To log your learning rate,  you can simply add `pl.callbacks.LearningRateMonitor(logging_interval=...)` to the list you pass to the `callbacks` argument of your trainer :
```
lr_monitor = pl.callbacks.LearningRateMonitor(logging_interval=lr_schedule_unit)
callbacks = [lr_monitor, ...]
trainer = pl.Trainer(callbacks = callbacks, ...)
trainer.fit(...)
```
